shm/alcohol.cpp

```diff
@@ -0,0 +1,17 @@
+#include "alcohol.hpp"
+
+Alcohol::Alcohol(std::string name, size_t amount, size_t alcoholByVolume)
+    : Cargo(name, amount, alcoholByVolume_){};
+
+bool Alcohol::operator==(Cargo &cargo) const {}
+
+size_t Alcohol::getPrice() const {
+  if (alcoholByVolume_ < maxAlcoholByVolume_) {
+    return basePrice_ * alcoholByVolume_ / maxAlcoholByVolume_;
+  }
+  return basePrice_;
+}
+
+std::string Alcohol::getName() const { return name_; }
+size_t getAmount() const { return amount_; }
```

Comment:
That part was a tricky one. I was curious if you notice, that `Alcohol` `Item` and `Fruit` have the same implementations for getName() and getAmount(). So please move this to the `Cargo` class. You can leave it virtual. But implementation is common for all classes derived from cargo.